mkdocs.yml

```diff
@@ -40,8 +40,33 @@ theme:
 plugins:
   - awesome-pages
   - search
+  - macros:
+      module_name: docs/scripts/doc_macros
+  - mkdocstrings:
```

Comment:
> Automated code API documentation. [Google's docstring style](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings) should be used.

This has significantly increased the `mkdocs serve` time to refresh or just being useable? Would be possible to have this as optional while writing docs. Or provide an alternative to preview docs quickly?

Reply:
Looking deeper this seems to be easy doable if the `gen_ref_pages.py` uses a environment variable based flag to generate the initial reference files. Then enable that during build only by default. If someone want to enable during `mkdocs serve` it should be posssible too. 

```
GEN_REF_PAGES=True mkdocs build        # Generate reference docs, use in CI/CD
```

```
mkdocs serve        # skip reference docs, to be used locally / `Makefile` 
```

Reply:
I also added an env check to gen_ref_files as that was still running w/o mkdocstrings being enabled.